### DISCUSSION TITLE
How to use for Graph Clustering.

### DISCUSSION THREAD

**Discussion by PeterDeSOM** [pyg-team/pytorch_geometric#7761] [Q&A]
I'm trying to classify graph data with fully unsupervised Graph Clustering.

So far, all that I've found the method for Graph Clustering is actually for node clustering or not a fully unsupervised learning method. It means that they eventually need their label to train at the downstream task.

For example, it is;
- pretraining some datasets with the unlabeled using the contrastive learning method or others. (No output channel or number of classes to be clustered is required)
- then finetuning the dataset with its labels using the supervised learning method.

I'm finding a fully unsupervised learning method for Graph-Level Clustering/Classification that output channel is required. Or, I'd like to know how to employ the typical fully unsupervised learning method to classify lots of graph data that cannot be labeled by human efforts.

Also, I want to employ DMoNPool on that fully unsupervised graph-level clustering.

Is it possible and could you show me a little direction how to?

Thank you

**Comment by rusty1s** [comment]
Can you share some insights why you don't think that common graph clustering techniques are not applicable for your use-case? Note that supervised learning based on the node embeddings generated by an unsupervised learning approach is commonly done to judge the quality of the embeddings - it is not required to obtain the clustering in the first place.
